diff --git a/android/src/newarch/java/com/credentialsmanager/CredentialsManagerModule.kt b/android/src/newarch/java/com/credentialsmanager/CredentialsManagerModule.kt
index 664814d3cc21b2145eb8ebbf5afb405364f93c1a..ee26d4b90efa5c1d54fa49fd6f98fe807f89d2fb 100644
--- a/android/src/newarch/java/com/credentialsmanager/CredentialsManagerModule.kt
+++ b/android/src/newarch/java/com/credentialsmanager/CredentialsManagerModule.kt
@@ -6,6 +6,7 @@ import androidx.credentials.exceptions.GetCredentialException
 import androidx.credentials.exceptions.NoCredentialException
 import com.credentialsmanager.handlers.CredentialHandler
 import com.credentialsmanager.handlers.ErrorHandler
+import com.facebook.react.bridge.Arguments
 import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.bridge.ReadableArray
@@ -68,11 +69,11 @@ class CredentialsManagerModule(
         credentialHandler.createPassword(username, password)
         
         // Create success response
-        val result = mapOf(
-          "type" to "password",
-          "username" to username,
-          "success" to true
-        )
+        val result = Arguments.createMap().apply {
+          putString("type", "password")
+          putString("username", username)
+          putBoolean("success", true)
+        }
         promise.resolve(result)
       } catch (e: CreateCredentialException) {
         ErrorHandler.handleCredentialError(e)
diff --git a/android/src/oldarch/java/com/credentialsmanager/CredentialsManagerModule.kt b/android/src/oldarch/java/com/credentialsmanager/CredentialsManagerModule.kt
index e35cdb362f1f7fcbdcab42185140942e7f424c32..581dac183289acf5e5b7e218cda281455e50bb80 100644
--- a/android/src/oldarch/java/com/credentialsmanager/CredentialsManagerModule.kt
+++ b/android/src/oldarch/java/com/credentialsmanager/CredentialsManagerModule.kt
@@ -6,6 +6,7 @@ import androidx.credentials.exceptions.GetCredentialException
 import androidx.credentials.exceptions.NoCredentialException
 import com.credentialsmanager.handlers.CredentialHandler
 import com.credentialsmanager.handlers.ErrorHandler
+import com.facebook.react.bridge.Arguments
 import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.bridge.ReactContextBaseJavaModule
@@ -74,11 +75,11 @@ class CredentialsManagerModule(
         credentialHandler.createPassword(username, password)
         
         // Create success response
-        val result = mapOf(
-          "type" to "password",
-          "username" to username,
-          "success" to true
-        )
+        val result = Arguments.createMap().apply {
+          putString("type", "password")
+          putString("username", username)
+          putBoolean("success", true)
+        }
         promise.resolve(result)
       } catch (e: CreateCredentialException) {
         ErrorHandler.handleCredentialError(e)
